cg/meta/upload/gisaid/gisaid.py

```diff
@@ -114,92 +142,117 @@ def get_gisaid_samples(self, case_id: str) -> list[GisaidSample]:
+        with open(gisaid_fasta_file, "w") as write_handle:
+            write_handle.writelines(fasta_lines)
 
-        if gisaid_fasta_file:
+        if gisaid_fasta:
             return
 
         self.housekeeper_api.add_and_include_file_to_latest_version(
-            bundle_name=case_id, file=gisaid_fasta_path, tags=["gisaid-fasta", case_id]
+            bundle_name=case_id, file=gisaid_fasta_file, tags=[GisaidTag.FASTA, case_id]
         )
 
-    def create_gisaid_csv(self, gisaid_samples: list[GisaidSample], case_id: str) -> None:
-        """Create csv file for gisaid upload"""
-        samples_df = pd.DataFrame(
-            data=[gisaid_sample.model_dump() for gisaid_sample in gisaid_samples],
-            columns=HEADERS,
+    def create_and_include_gisaid_samples_csv_to_hk(
+        self, gisaid_samples: list[GisaidSample], case_id: str
+    ) -> None:
+        """Create and include CSV file for GISAID samples to Housekeeper."""
+        gisaid_samples_csv: File | None = self.housekeeper_api.get_file_from_latest_version(
+            bundle_name=case_id, tags={GisaidTag.CSV, case_id}
         )
-
-        gisaid_csv_file = self.housekeeper_api.get_file_from_latest_version(
-            bundle_name=case_id, tags=["gisaid-csv", case_id]
-        )
-        if gisaid_csv_file:
-            LOG.info(f"GISAID CSV for case {case_id} exists, will be replaced")
-            gisaid_csv_path = gisaid_csv_file.full_path
+        if gisaid_samples_csv:
+            LOG.info(f"GISAID samples CSV for case {case_id} exists, and will be replaced")
+            gisaid_samples_csv_file = Path(gisaid_samples_csv.full_path)
         else:
-            gisaid_csv_path = self.get_gisaid_csv_path(case_id=case_id)
-        samples_df.to_csv(gisaid_csv_path, sep=",", index=False)
-
-        if gisaid_csv_file:
+            gisaid_samples_csv_file: Path = self.get_gisaid_sample_csv_path(case_id=case_id)
+        samples: list[dict] = [sample.model_dump() for sample in gisaid_samples]
+        write_csv_from_dict(content=samples, fieldnames=HEADERS, file_path=gisaid_samples_csv_file)
+        if gisaid_samples_csv:
             return
-
         self.housekeeper_api.add_and_include_file_to_latest_version(
-            bundle_name=case_id, file=gisaid_csv_path, tags=["gisaid-csv", case_id]
+            bundle_name=case_id, file=gisaid_samples_csv_file, tags=[GisaidTag.CSV, case_id]
         )
 
-    def create_gisaid_log_file(self, case_id: str) -> None:
-        """Path for gisaid bundle log"""
-        gisaid_log_file = self.housekeeper_api.get_files(
-            bundle=case_id, tags=["gisaid-log", case_id]
-        ).first()
-        if gisaid_log_file:
-            LOG.info("GISAID log exists in case bundle in Housekeeper")
+    def create_and_include_gisaid_log_file_to_hk(self, case_id: str) -> None:
+        """Create and include GISAID log file to a Housekeeper."""
+        if _ := self.housekeeper_api.get_files(
```

Comment:
I don't think this walrus operator is needed here if the value is not going to be used
```suggestion
        if self.housekeeper_api.get_files(
```